sys/dev/isp/isp.c

```diff
@@ -432,11 +559,28 @@ isp_reset(ispsoftc_t *isp, int do_load_defaults)
 	MBSINIT(&mbs, MBOX_EXEC_FIRMWARE, MBLOGALL, 5000000);
 	mbs.param[1] = code_org >> 16;
 	mbs.param[2] = code_org;
-	if (!IS_26XX(isp))
-		mbs.param[3] = loaded_fw ? 0 : 1;
+	mbs.param[3] = loaded_fw ? 0 : 1;
+	mbs.param[4] = 0;
+	mbs.param[11] = 0;
+	if (IS_27XX(isp))
+		mbs.param[4] |= 0x08;	/* NVME_ENABLE_FLAG */
 	isp_mboxcmd(isp, &mbs);
 	if (mbs.param[0] != MBOX_COMMAND_COMPLETE)
 		return;
+	fcp->fw_ability_mask = (mbs.param[3] << 16) | mbs.param[2];
+	isp_prt(isp, ISP_LOGDEBUG0, "Firmware ability mask: 0x%x",
+	    fcp->fw_ability_mask);
+	if (IS_27XX(isp)) {
```

Comment:
Why not IS_26XX() here?  My 26xx+27xx specs do mention it.

Reply:
Again, no specs here just the Linux driver.
```
if (IS_QLA27XX(ha) || IS_QLA28XX(ha)) {
                ha->max_supported_speed = mcp->mb[2] & (BIT_0|BIT_1);
                ql_dbg(ql_dbg_mbx, vha, 0x119b, "max_supported_speed=%s.\n",
....
```
So I assumed that only 27xx+ support reporting the max supported speed.

Reply:
Tested this with a 26xx based adpater and you (and the specs) are right.
Changed it to `IS_26XX()`